### PR TITLE
Premultiplied view*proj matrix 

### PR DESCRIPTION
**Pull request by nico-abram**
## Description

Attempt to do https://github.com/amethyst/amethyst/issues/1902

## Modifications

- Add a view_proj element to what's passed to the vertex shader and use it instead of doing view * proj

I can squash commits in any combination needed. I only tested this a bit on windows with the vulkan feature/backend.

### CODE REVIEW

**Comment by nico-abram** [comment]
Thanks for the response.

I wasn't sure if such a small change merited a changelog entry, just did it :+1:

File: docs/CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,7 @@ The format is based on [Keep a Changelog][kc], and this project adheres to
 ### Changed
 
 * Updated `syn`, `quote`, and `proc-macro2` to `1.0`. ([#1952])
+* Use a premultiplied view_proj matrix in vertex shaders. ([#1964])
```
Comment (distransient) [inline docs/CHANGELOG.md]: Sorry, you just added your changelog entry to the 0.13.1 section. You'll have to rebase and add it to the new Unreleased section.

Comment (nico-abram) [reply docs/CHANGELOG.md]: Done! Sorry about that.

Comment (distransient) [reply docs/CHANGELOG.md]: Sorry, I broke it again :P emergency patch release happened last night

Comment (nico-abram) [reply docs/CHANGELOG.md]: No problem, done!

**Comment by jaynus** [comment]
Looks like this conflicts on the changelog again, but besides that I approve.

I was thinking about this needing projview * model precalculated, but I think the way you did it is better. Projview is basically once-per-pass at the moment, while model would be per entity. I believe it's faster to do that on the shader since I don't trust nalgebra

**Comment by jaynus** [comment]
We store the compile spv to prevent dependencies on shaderc by default. Could you please run 'make' in the root folder so we get new prebuilt shaders and so you can actually test this?

**Comment by azriel91** [comment]
bors r=azriel91, distransient, jaynus

**Comment by jaynus** [comment]
You are so judgemental.

Bors retry

**Comment by jaynus** [comment]
You were fat bors retry

**Comment by jaynus** [comment]
Bors retry

**Pull request merged**